From afa0d2f7e85c7290c84b215f8aaf27a226959ca3 Mon Sep 17 00:00:00 2001
From: Xinhui Yang <cyan@cyano.uk>
Date: Wed, 17 Sep 2025 13:23:36 +0800
Subject: [PATCH] ia64: remove libunwind symbols in libgcc

The test config.host mentioned `use_system_libunwind' is long gone,
besides, libunwind has been moved to a separate project, and IA-64
support is still presents. For any shared libraries they can depend on
this libunwind if they want to.

Additionally, this was creating problems while bootstrapping a cross
toolchain where the static libgcc.a contains versioned symbols that
references corresponding symbols in libunwind, causing any shared
libraries that wants to link with libgcc.a to fail with a "version
node not found for symbol: _Unwind_*@@GCC_3.2.0" message.

libgcc/
	* config.host: Remove ia64/t-linux-libunwind; Remove
	  `use_system_libunwind' test.

* libgcc/config/ia64
	* t-linux: Ditch unwind-compat.c.
	* t-linux-libunwind: Remove.
---
 libgcc/config.host                   | 5 +----
 libgcc/config/ia64/t-linux           | 4 ++--
 libgcc/config/ia64/t-linux-libunwind | 3 ---
 3 files changed, 3 insertions(+), 9 deletions(-)
 delete mode 100644 libgcc/config/ia64/t-linux-libunwind

diff --git a/libgcc/config.host b/libgcc/config.host
index 6a88ee5a2dd..4e63edfc00b 100644
--- a/libgcc/config.host
+++ b/libgcc/config.host
@@ -967,10 +967,7 @@ ia64*-*-freebsd*)
 ia64*-*-linux*)
 	# Don't use crtbeginT.o from *-*-linux* default.
 	extra_parts="crtbegin.o crtend.o crtbeginS.o crtendS.o crtfastmath.o"
-	tmake_file="$tmake_file ia64/t-ia64 ia64/t-ia64-elf t-crtfm t-softfp-tf ia64/t-softfp t-softfp ia64/t-softfp-compat ia64/t-eh-ia64 t-libunwind ia64/t-linux"
-	if test x$with_system_libunwind != xyes ; then
-		tmake_file="${tmake_file} t-libunwind-elf ia64/t-linux-libunwind"
-	fi
+	tmake_file="$tmake_file ia64/t-ia64 ia64/t-ia64-elf t-crtfm t-softfp-tf ia64/t-softfp t-softfp ia64/t-softfp-compat ia64/t-eh-ia64 ia64/t-linux"
 	md_unwind_header=ia64/linux-unwind.h
 	;;
 ia64*-*-hpux*)
diff --git a/libgcc/config/ia64/t-linux b/libgcc/config/ia64/t-linux
index e6d72b94a87..3e99efb6ff3 100644
--- a/libgcc/config/ia64/t-linux
+++ b/libgcc/config/ia64/t-linux
@@ -1,5 +1,5 @@
 # Use system libunwind library on IA-64 GLIBC based system.
-LIB2ADDEH = $(srcdir)/unwind-sjlj.c $(srcdir)/unwind-c.c \
-  $(srcdir)/unwind-compat.c
+# libunwind does not have SjTj handling.
+LIB2ADDEH = $(srcdir)/unwind-sjlj.c $(srcdir)/unwind-c.c
 
 SHLIB_MAPFILES += $(srcdir)/config/ia64/libgcc-glibc.ver
diff --git a/libgcc/config/ia64/t-linux-libunwind b/libgcc/config/ia64/t-linux-libunwind
deleted file mode 100644
index 8b1736a2d67..00000000000
--- a/libgcc/config/ia64/t-linux-libunwind
+++ /dev/null
@@ -1,3 +0,0 @@
-# Build libunwind for IA-64 GLIBC based system.
-LIBUNWIND = $(srcdir)/config/ia64/fde-glibc.c \
-  $(srcdir)/config/ia64/unwind-ia64.c
-- 
2.51.0

